### DISCUSSION TITLE
Applying database migrations failed. This is mostly caused by the database being unavailable.

### DISCUSSION THREAD

**Discussion by RLDiary** [langfuse/langfuse#1336] [Support]
Hiya,

Been trying to fix this error for a couple of days now. I am trying to deploy the application on Amazon EC2 Linux 2. The Postgresql14.10 database is accepting connections "localhost:5432 - accepting connections". I have tested the connection through psycopg2 on Python and I have no trouble connecting to it.

However, when I try to run the docker run code as below, I keep getting error.

> $ docker run --name langfuse \
> -e DATABASE_URL=postgresql://username:password@localhost:5432/langfuse \
> -e NEXTAUTH_URL=http://localhost:3000 \
> -e NEXTAUTH_SECRET=**** \
> -e SALT=**** \
> -p 3000:3000 \
> -a STDOUT \
> ghcr.io/langfuse/langfuse:latest

> Prisma schema loaded from prisma/schema.prisma
> Datasource "db": PostgreSQL database "langfuse", schema "public" at "localhost:5432"

> Applying database migrations failed. This is mostly caused by the database being unavailable.
> Exiting...

I imagine this is because prisma migrate deploy fails. I have no idea how to fix it. Any advice on how I can fix this please?

**Comment by RLDiary** [comment]
Rookier mistake. Please ignore.

 PostgreSQL was not accessible from the Docker container. Replacing 'localhost' with docker's default bridge network 172.17.0.0 fixed the problem. Thank you ChatGPT!

**Comment by marcklingen** [comment]
thanks for confirming, this would have been a P0 issue otherwise :)
